turbinia/client.py

```diff
@@ -166,6 +167,90 @@ def __init__(self):
     self.task_manager = task_manager.get_task_manager()
     self.task_manager.setup(server=False)
 
+  def _create_default_recipe(self):
+    """Creates a default Turbinia recipe."""
+    default_recipe = recipe_helpers.DEFAULT_RECIPE
+    return default_recipe
+
+  def create_recipe(
+      self, debug_tasks=False, filter_patterns=None, group_id='',
+      jobs_allowlist=None, jobs_denylist=None, recipe_name=None, sketch_id=None,
+      skip_recipe_validation=False, yara_rules=None):
+    """Creates a Turbinia recipe. If no recipe_name is not specified, this
+        method returns a default recipe. If a recipe_name is specified
+        then this method will build the recipe dictionary by reading the
+        contents of a recipe file. The path to the recipe file is inferred
+        from the recipe_name and the RECIPE_FILE_DIR configuration
+        parameter.
+
+    Args:
+      debug_tasks (bool): flag to turn debug output on for supported tasks.
+      filter_patterns (list): a list of filter pattern strings.
+      group_id (str): a group identifier.
+      jobs_allowlist (list): a list of jobs allowed for execution.
+      jobs_denylist (list): a list of jobs that will not be executed.
+      recipe_name (str): Turbinia recipe name (e.g. triage-linux).
+      sketch_id (str): a Timesketch sketch identifier.
+      skip_recipe_validation (bool): flag indicates if the recipe will be
+          validated.
+      yara_rules (str): a string containing yara rules.
+
+    Returns:
+      dict: a Turbinia recipe dictionary.
+    """
+    recipe = None
+    if jobs_allowlist and jobs_denylist:
+      raise TurbiniaException(
+          'jobs_allowlist and jobs_denylist are mutually exclusive.')
+
+    # if no recipe_name is given, create a default recipe.
+    if not recipe_name:
+      recipe = self._create_default_recipe()
+      if filter_patterns:
+        recipe['globals']['filter_patterns'] = filter_patterns
+      if jobs_denylist:
+        recipe['globals']['jobs_denylist'] = jobs_denylist
+      if jobs_allowlist:
+        recipe['globals']['jobs_allowlist'] = jobs_allowlist
+      if yara_rules:
+        recipe['globals']['yara_rules'] = yara_rules
+    else:
+      # Load custom recipe from given path or name.
+      if '/' in recipe_name and recipe_name.endswith('.yaml'):
```

Comment:
Maybe `os.path.exists()` would be a better check here?

Reply:
Done